<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

  <xacro:macro name="aubo_ros2_control" params="name prefix robot_ip use_mock_hardware:=false mock_sensor_commands:=false">

    <ros2_control name="${name}" type="system">

      <hardware>
        <xacro:if value="${use_mock_hardware}">
          <plugin>mock_components/GenericSystem</plugin>
          <param name="mock_sensor_commands">${mock_sensor_commands}</param>
          <param name="state_following_offset">0.0</param>
        </xacro:if>
        <xacro:unless value="${use_mock_hardware}">
          <plugin>aubo_driver/AuboPositionHardwareInterface</plugin>
          <param name="robot_ip">${robot_ip}</param>
          <param name="servoj_frequency">80</param>
          <param name="servoj_smooth_scale">1.0</param>
          <param name="servoj_delay_scale">1.0</param>
          <param name="servoj_buffer_time">0.01</param>
          <param name="max_vel_limit">3.1</param>
          <param name="max_acc_limit">7.0</param>
          <param name="max_jerk_limit">200.0</param>
        </xacro:unless>
      </hardware>

      <joint name="${prefix}shoulder_pan_joint">
        <command_interface name="position">
          <param name="min">-3.04</param>
          <param name="max">3.04</param>
        </command_interface>
        <state_interface name="position"/>
      </joint>
      <joint name="${prefix}shoulder_lift_joint">
        <command_interface name="position">
          <param name="min">-3.04</param>
          <param name="max">3.04</param>
        </command_interface>
        <state_interface name="position"/>
      </joint>
      <joint name="${prefix}elbow_joint">
        <command_interface name="position">
          <param name="min">-3.04</param>
          <param name="max">3.04</param>
        </command_interface>
        <state_interface name="position"/>
      </joint>
      <joint name="${prefix}wrist_1_joint">
        <command_interface name="position">
          <param name="min">-3.04</param>
          <param name="max">3.04</param>
        </command_interface>
        <state_interface name="position"/>
      </joint>
      <joint name="${prefix}wrist_2_joint">
        <command_interface name="position">
          <param name="min">-3.04</param>
          <param name="max">3.04</param>
        </command_interface>
        <state_interface name="position"/>
      </joint>
      <joint name="${prefix}wrist_3_joint">
        <command_interface name="position">
          <param name="min">-3.04</param>
          <param name="max">3.04</param>
        </command_interface>
        <state_interface name="position"/>
      </joint>
    </ros2_control>

  </xacro:macro>

</robot>